var/spack/repos/builtin/packages/oce/package.py

```diff
@@ -87,7 +87,11 @@ def install(self, spec, prefix):
                 '-DOCE_OSX_USE_COCOA:BOOL=ON',
             ])
 
-        options.append('-DCMAKE_INSTALL_NAME_DIR:PATH=%s/lib' % prefix)
+        if '.'.join(platform.mac_ver()[0].split('.')[:2]) == '10.12':
+            # use @rpath on Sierra due to limit of dynamic loader
+            options.append('-DCMAKE_MACOSX_RPATH=ON')
```

Comment:
@mathstuf @davydden @alalazo: is this something we should be adding to all `CMakePackage`s?  Or is it even something we should handle below `CMake`, e.g. by adding some more compiler options?  Or is this only needed for CMake builds on OS X because CMake does its own weird RPATH handling?